Local fields are created unnecessarily instead of using existing ones in superclass

We shouldn't need to define a new local field in `B` here:
```scala
class A(val aaa: Int)
class B(val aaa: Int) extends A(aaa)
```
But we do in dotty:
```scala
package <empty> {
  class A extends Object { 
    def <init>(aaa: Int): A = {
      super()
      this.aaa$$local = aaa
      ()
    }
    private val aaa$$local: Int
    <accessor> def aaa(): Int = this.aaa$$local
  }
  final lazy module val A: A$ = new A$()
  final module class A$ extends Object { this: A$ => 
    def <init>(): A$ = {
      super()
      ()
    }
  }
  class B extends A { 
    def <init>(aaa: Int): B = {
      super(aaa)
      this.aaa$$local = aaa
      ()
    }
    private val aaa$$local: Int
    override <accessor> def aaa(): Int = this.aaa$$local
    def foo(): Int = B.this.aaa().+(1)
  }
  final lazy module val B: B$ = new B$()
  final module class B$ extends Object { this: B$ => 
    def <init>(): B$ = {
      super()
      ()
    }
  }
}
```
Compare with scalac:
```scala
package <empty> {
  class A extends Object {
    <paramaccessor> private[this] val aaa: Int = _;
    <stable> <accessor> <paramaccessor> def aaa(): Int = A.this.aaa;
    def <init>(aaa: Int): A = {
      A.this.aaa = aaa;
      A.super.<init>();
      ()
    }
  };
  class B extends A {
    override <stable> <accessor> <paramaccessor> def aaa(): Int = (B.super.aaa(): Int);
    def foo(): Int = B.this.aaa().+(1);
    def <init>(aaa: Int): B = {
      B.super.<init>(aaa);
      ()
    }
  }
}
```

I have a fix with an extensive test case at https://github.com/smarter/dotty/commits/fix/param-forwarding unfortunately the following currently fail:
```scala
class A(val aaa: Int)
class B(override val aaa: Int) extends A(aaa)
```
With:
```scala
try/superarg2.scala:2: error: overriding value aaa in class A of type Int;
 method aaa of type => Int needs to be a stable, immutable value
class B(override val aaa: Int) extends A(aaa)
```
The failure comes from `RefChecks`:
```scala
      } else if (other.isStable && !member.isStable) { // (1.4)
        overrideError("needs to be a stable, immutable value")
```
Because at the point where `RefChecks` is run, the tree looks like:
```scala
class A(aaa: Int) extends Object() {
  val aaa: Int
}
class B(aaa: Int) extends A(aaa) {
  override def aaa: Int = super.aaa
}
```

What would be the best way to avoid this? Should I try using `@uncheckedStable`?

--- a/src/dotty/tools/dotc/transform/ParamForwarding.scala
+++ b/src/dotty/tools/dotc/transform/ParamForwarding.scala
@@ -43,13 +43,13 @@ class ParamForwarding(thisTransformer: DenotTransformer) {
         stat match {
           case stat: ValDef =>
             val sym = stat.symbol.asTerm
-            if (sym is (PrivateLocalParamAccessor, butNot = Mutable)) {
+            if (sym is (ParamAccessor, butNot = Mutable)) {
               val idx = superArgs.indexWhere(_.symbol == sym)
               if (idx >= 0 && superParamNames(idx) == stat.name) { // supercall to like-named parameter
                 val alias = inheritedAccessor(sym)
                 if (alias.exists) {
                   def forwarder(implicit ctx: Context) = {
-                    sym.copySymDenotation(initFlags = sym.flags | Method, info = sym.info.ensureMethodic)
+                    sym.copySymDenotation(initFlags = sym.flags | Method | Stable, info = sym.info.ensureMethodic)
                       .installAfter(thisTransformer)
                     val superAcc =
                       Super(This(currentClass), tpnme.EMPTY, inConstrCall = false).select(alias)

--- a/tests/run/paramForwarding.check
+++ b/tests/run/paramForwarding.check
@@ -1,0 +1,16 @@
+B:
+private final int B.theValue$$local
+private final int B.theValueInB$$local
+Bz:
+private final int Bz.theValue$$local
+private final int Bz.theValueInBz$$local
+C:
+
+D:
+private final int D.other$$local
+NonVal:
+
+X:
+private final int X.theValue$$local
+Y:
+private final int Y.theValue$$local

--- a/tests/run/paramForwarding.scala
+++ b/tests/run/paramForwarding.scala
@@ -1,0 +1,89 @@
+// A contains a field A.theValue$$local accessible using the
+// generated getter A.theValue()
+class A(val theValue: Int) {
+  val theValueInA = theValue // use the constructor parameter theValue
+
+  def getTheValue = theValue // virtual call to the getter theValue()
+}
+
+// B contains a field B.theValue$$local accessible using the getter
+// B.theValue() which overrides A.theValue()
+class B(override val theValue: Int) extends A(42) {
+  val theValueInB = theValue
+}
+
+// Bz contains a field Bz.theValue$$local accessible using the getter
+// Bz.theValue() which overrides A.theValue()
+class Bz extends A(42) {
+  override val theValue: Int = 10
+  val theValueInBz = theValue
+}
+
+// C does not contains a field C.theValue$$local, it contains
+// a getter C.theValue() which only calls super.theValue()
+class C(override val theValue: Int) extends A(theValue)
+
+// D contains a field D.other$$local and a corresponding getter.
+class D(val other: Int) extends A(other)
+
+
+// NonVal does not contain a field NonVal.theValue$$local.
+class NonVal(theValue: Int) extends A(theValue) {
+  def getTheValueInNonVal = theValue // use the constructor parameter theValue
+}
+
+// X contains a field X.theValue$$local accessible using the getter
+// X.theValue() which overrides A.theValue()
+class X(override val theValue: Int) extends NonVal(0)
+
+// Y contains a field Y.theValue$$local accessible using the getter
+// Y.theValue() which overrides A.theValue()
+class Y(override val theValue: Int) extends NonVal(theValue)
+
+
+object Test {
+  def printFields(obj: Any) =
+    println(obj.getClass.getDeclaredFields.map(_.toString).sorted.deep.mkString("\n"))
+
+  def main(args: Array[String]): Unit = {
+    val b10 = new B(10)
+    val bz = new Bz
+    val c11 = new C(11)
+    val d12 = new D(12)
+    val nv13 = new NonVal(13)
+    val x14 = new X(14)
+    val y15 = new Y(15)
+
+    println("B:")
+    printFields(b10)
+    println("Bz:")
+    printFields(bz)
+    println("C:")
+    printFields(c11)
+    println("D:")
+    printFields(d12)
+    println("NonVal:")
+    printFields(nv13)
+    println("X:")
+    printFields(x14)
+    println("Y:")
+    printFields(y15)
+
+
+    assert(b10.getTheValue == 10)
+    assert(b10.theValue == 10)
+    assert(b10.theValueInB == 10)
+    assert(b10.theValueInA == 42)
+
+    assert(bz.getTheValue == 10)
+    assert(bz.theValue == 10)
+    assert(bz.theValueInBz == 10)
+    assert(bz.theValueInA == 42)
+
+
+    assert(x14.theValue == 14)
+    assert(x14.getTheValue == 14)
+    assert(x14.getTheValueInNonVal == 0)
+    assert(x14.theValueInA == 0)
+  }
+}

